### DISCUSSION TITLE
Experimental react hook `useOptimistic`

### DISCUSSION THREAD

**Discussion by ryparker** [47ng/nuqs#515] [General] [answered]
Found an article that might be of interest to this project in the future (once the `useOptimistic` React hook is stable). 

The article is about using the `useOptimistic` hook to improve search param usage in server components. The hook allows for the merging of ephemeral client-side state with server-side data, making it ideal for creating filter panels that react instantly to user interactions while relying on the URL as the source of truth.

Article: https://buildui.com/posts/instant-search-params-with-react-server-components
React docs for `useOptimistic`: https://react.dev/reference/react/useOptimistic

**Comment by franky47** [comment, marked as answer]
I replied to Sam's post and discussed the idea: https://twitter.com/fortysevenfx/status/1762573338323333493

One issue I have with the described approach is  that the URL only shows the change once the RSC payload has been received. `nuqs` favours a reactive URL update, and notifies the server asynchronously in the background, without blocking.
